src/Fixer/PhpUnit/PhpUnitDedicateAssertFixer.php

```diff
@@ -178,23 +178,37 @@ public function getPriority()
      */
     protected function applyFix(\SplFileInfo $file, Tokens $tokens)
     {
-        static $searchSequence = [
-            [T_VARIABLE, '$this'],
-            [T_OBJECT_OPERATOR, '->'],
-            [T_STRING],
+        static $searchSequences = [
+            [
+                [T_VARIABLE, '$this'],
+                [T_OBJECT_OPERATOR, '->'],
+                [T_STRING],
+            ],
+            [
+                [T_STATIC, 'static'],
+                [T_DOUBLE_COLON, '::'],
+                [T_STRING],
+            ],
+            [
+                [T_STRING, 'self'],
+                [T_DOUBLE_COLON, '::'],
+                [T_STRING],
+            ],
         ];
 
-        $index = 1;
-        $candidate = $tokens->findSequence($searchSequence, $index);
-        while (null !== $candidate) {
-            end($candidate);
-            $index = $this->getAssertCandidate($tokens, key($candidate));
-            if (is_array($index)) {
-                $index = $this->fixAssert($tokens, $index);
-            }
-
-            ++$index;
+        foreach ($searchSequences as $searchSequence) {
+            $index = 1;
             $candidate = $tokens->findSequence($searchSequence, $index);
```

Comment:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3477#discussion_r164154150

Reply:
ping @Slamdunk , could you please take a look at that ? :)
we don't want to slower this rule by 3 times

Reply:
Sorry for the late reply: this isn't strictly as slow as https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3477#discussion_r164154150 I don't think we can improve much the performance without making this method much more complex and unreadable.

Reply:
well, this makes the fixer to run over whole tokens collection not once, but 3 times, isn't it ?

Reply:
Algorithm optimized

Reply:
> i would propose to drop findSequence at all here, iterate tokens manually, search for T_STRING matching the assertion, if it does then ensure prev and prev-prev tokens are matching $this->/self::/static:: and continue with current fixing logic

I've applied this algorithm before reading your modified comment, and I wrote it like you described. Please stop reading out mind, I'm scared you'll start fixing our thoughts too.

Reply:
:D
well, I'm proud of fixing few guys thoughts already, you know, fixing the initial mindset "I don't need tests" :)